ignition-server/controllers/machineconfigserver_ignitionprovider.go

```diff
@@ -215,17 +237,24 @@ cp /assets/manifests/*.machineconfigpool.yaml /mcc-manifests/bootstrap/manifests
 	customMachineConfigArg := `
 cat /tmp/custom-config/base64CompressedConfig | base64 -d | gunzip --force --stdout > /mcc-manifests/bootstrap/manifests/custom.yaml`
 
+	podName := fmt.Sprintf("%s%d", resourceGenerateName, rand.Int31())
 	return &corev1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace:    namespace,
-			GenerateName: resourceGenerateName,
+			Namespace: namespace,
+			Name:      podName,
+			Labels: map[string]string{
+				"app": "machine-config-server",
+			},
 		},
 		Spec: corev1.PodSpec{
 			ServiceAccountName:            sa.Name,
 			TerminationGracePeriodSeconds: k8sutilspointer.Int64Ptr(10),
+			EnableServiceLinks:            k8sutilspointer.BoolPtr(true),
+			Subdomain:                     mcsPodSubdomain,
+			Hostname:                      podName,
 			Tolerations: []corev1.Toleration{
 				{
-					Key:      "multi-az-worker",
+					Key:      "hypershift.openshift.io/cluster",
```

Comment:
I don't think the value for this key is ever "true" but the cluster key (hcp namespace) https://github.com/openshift/hypershift/blob/bd0715270c7a2646e4339f2222d210237b05f39d/control-plane-operator/controllers/hostedcontrolplane/config/deployment.go#L121-L123

So we should either drop this as it has no effect or considering passing the cluster key (hcp namespace) through as possibly token Secret new key/value.

Reply:
First iteration I believe Sanjeev has dropped and we will follow up with a todo to get it proper with tolerations